### ISSUE TITLE
Optional function parameters should not be considered a "branch"

### ISSUE THREAD

**Issue by MaxBeauchemin** [Kotlin/kotlinx-kover#303]
**What is your use-case and why do you need this feature?**

Function: `fun foo(bar: Boolean = false) = if (bar) "bar" else "foobar"`

Invocation: `foo()` is missing coverage since it's utilizing the default parameter

This is **especially** important when the function that has optional parameters is not one that we wrote or have control over

i.e. `kotlinx.coroutines.flow.toList` is giving us this issue today

![image](https://user-images.githubusercontent.com/12040012/218833823-57b52768-0270-4132-9755-14ad7eab292f.png)

**Describe the solution you'd like**

Optional parameters should not be considered a branching condition, otherwise we can't leverage them without missing coverage. 

(A setting to toggle this feature would also be sufficient if there are projects that like the current behavior)

**Comment by shanshin** [comment]
Hi, 
It is difficult to understand the reason for incomplete coverage from the screenshot.

However, it is not related to the default argument, because for a function that is not fully covered, coverage problems are displayed at the place of definition of this function, and not at call side.

In your example, for the function `fun foo(bar: Boolean = false) = if (bar) "bar" else "foobar"` if you invoke
```
        foo(true)
        foo()
```
or 
```
        foo(true)
        foo(false)
```
this function will be fully covered in the report
![image](https://user-images.githubusercontent.com/7123498/218847289-eb9577e3-4a90-4710-84ae-1f41e7e92ba4.png)


For a more detailed learning of the issue, could you please send a binary coverage report (located in the build/kover/<launched_test_name>.ic directory)?

**Comment by MaxBeauchemin** [comment]
(for reference) This is the `toList()` function we're invoking (in the `kotlinx.coroutines.flow` package)

```
public suspend fun <T> Flow<T>.toList(destination: MutableList<T> = ArrayList()): List<T> = toCollection(destination)
```

**Comment by MaxBeauchemin** [comment]
> Hi, unfortunately, it is difficult to understand the reason for incomplete coverage by the screenshot.
> 
> However, it is not related to the default argument, because for a function that is not fully covered, coverage problems are displayed at the place of definition of this function, and not at call side.
> 
> In your example, for the function `fun foo(bar: Boolean = false) = if (bar) "bar" else "foobar"` if you invoke
> 
> ```
>         foo(true)
>         foo()
> ```
> 
> or
> 
> ```
>         foo(true)
>         foo(false)
> ```
> 
> this function will be fully covered in the report ![image](https://user-images.githubusercontent.com/7123498/218847289-eb9577e3-4a90-4710-84ae-1f41e7e92ba4.png)
> 
> For a more detailed learning of the issue, could you please send a binary coverage report (located in the build/kover/<launched_test_name>.ic directory)?

In reply to this, the line of code I'm trying to get covered is `foo()` not the actual function implementation itself

**Comment by shanshin** [comment]
> In reply to this, the line of code I'm trying to get covered is foo() not the actual function implementation itself

Default arguments can only affect the definition of a function.
In your case, the problem may be caused by an additional byte-code change by the coroutines.

**Comment by shanshin** [comment]
Fixed in `0.7.4`
